python/lsst/ts/atpneumaticssimulator/pneumatics_simulator.py

```diff
@@ -239,6 +387,179 @@ async def write_noack_response(self, sequence_id: int) -> None:
+            await self._write_evt(
+                evt_id=Event.CELLVENTSTATE, state=self.cell_vents_state
+            )
+
+        self.m1_vents_limit_switches.ventsClosedActive = closed
+        self.m1_vents_limit_switches.ventsOpenedActive = opened
+        await self._write_evt(
+            evt_id=Event.M1VENTSLIMITSWITCHES,
+            ventsClosedActive=self.m1_vents_limit_switches.ventsClosedActive,
+            ventsOpenedActive=self.m1_vents_limit_switches.ventsOpenedActive,
+        )
+        if opened:
+            self.m1_vents_position = ATPneumatics.VentsPosition.OPENED
+            self.cell_vents_state = ATPneumatics.CellVentState.OPENED
+            await self._write_evt(
+                evt_id=Event.CELLVENTSTATE, state=self.cell_vents_state
+            )
+        elif closed:
+            self.m1_vents_position = ATPneumatics.VentsPosition.CLOSED
+            self.cell_vents_state = ATPneumatics.CellVentState.CLOSED
+            await self._write_evt(
+                evt_id=Event.CELLVENTSTATE, state=self.cell_vents_state
+            )
+        else:
+            self.m1_vents_position = ATPneumatics.VentsPosition.PARTIALLYOPENED
+        await self._write_evt(
+            evt_id=Event.M1VENTSPOSITION, position=self.m1_vents_position
+        )
+
+    async def set_m1_cover_events(self, closed: bool, opened: bool) -> None:
+        """Set m1CoverLimitSwitches and m1CoverState events.
+
+        Output any changes.
+
+        Parameters
+        ----------
+        closed : `bool`
+            Are the closed switches active?
+        opened : `bool`
+            Are the opened switches active?
+        """
+        self.m1_cover_limit_switches.cover1ClosedActive = closed
+        self.m1_cover_limit_switches.cover2ClosedActive = closed
+        self.m1_cover_limit_switches.cover3ClosedActive = closed
+        self.m1_cover_limit_switches.cover4ClosedActive = closed
+        self.m1_cover_limit_switches.cover1OpenedActive = opened
+        self.m1_cover_limit_switches.cover2OpenedActive = opened
+        self.m1_cover_limit_switches.cover3OpenedActive = opened
+        self.m1_cover_limit_switches.cover4OpenedActive = opened
+        await self._write_evt(
+            evt_id=Event.M1COVERLIMITSWITCHES,
+            cover1ClosedActive=self.m1_cover_limit_switches.cover1ClosedActive,
+            cover2ClosedActive=self.m1_cover_limit_switches.cover2ClosedActive,
+            cover3ClosedActive=self.m1_cover_limit_switches.cover3ClosedActive,
+            cover4ClosedActive=self.m1_cover_limit_switches.cover4ClosedActive,
+            cover1OpenedActive=self.m1_cover_limit_switches.cover1OpenedActive,
+            cover2OpenedActive=self.m1_cover_limit_switches.cover2OpenedActive,
+            cover3OpenedActive=self.m1_cover_limit_switches.cover3OpenedActive,
+            cover4OpenedActive=self.m1_cover_limit_switches.cover4OpenedActive,
+        )
```

Comment:
Please consider a loop. It could both set the events and construct the kwargs for this call to `_write_evt`. E.g.:
```
kwargs = dict()
for num in range(1, 5):
    closed_attr_name = f"cover{num}ClosedActive"
    opened_attr_name = f"cover{num}OpenedActive"
    setattr(self.m1_cover_limit_switches, closed_attr_name, closed)
    setattr(self.m1_cover_limit_switches, opened_attr_name, opened)
    kwargs[closed_attr_name] = closed
    kwargs[opened_attr_name] = opened
await self._write_evt(evt_id= Event.M1COVERLIMITSWITCHES, **kwargs)
```